src/redux/slices/profileSlice.ts

```diff
@@ -15,11 +15,18 @@ export const profileSlice = createSlice({
   reducers: {
     setProfile: (state, action: PayloadAction<Profile>) => {
       const newProfile = action.payload;
-      state.profiles.set(newProfile.socialAddress, newProfile);
+      const oldProfile = state.profiles.find(
+        (profile) => profile.socialAddress === newProfile.socialAddress
+      );
+      if (oldProfile) state.profiles.splice(state.profiles.indexOf(oldProfile));
```

Comment:
do we want to remove a single item or multiple? Here I think splice is removing everything from the start.

Reply:
It says for splice:
`"Removes elements from an array and, if necessary, inserts new elements in their place".`
So if I was adding multiple profiles yeah it would remove multiple but the array `find` says:
`"Returns the value of the first element in the array where predicate is true, and undefined otherwise."`

So I believe we should only ever find a single profile with the same address and replace it. My understanding is that the DSNP identity address is unique, so this should work as intended. Let me know if there may be an issue with my line of thinking on it.

Reply:
Misread the question, I did make a mistake here, need to pass to splice the number of elements to remove starting at the passed index. Fixed, thanks!